File: DSA#/Sorting-Algos/BubbleSort.java
//Bubble Sort: Compare the adjacent elements and swap them if they are in wrong order repeatedly, until the array is sorted.

public class BubbleSort {
    public static void bubbleSort(int[] arr){
        int n =arr.length;
        for(int i=0; i<n-1; i++){
            for(int j=0; j<n-i-1; j++){
                if(arr[j]>arr[j+1]){
                    //Swap using a temp variable
                    int temp=arr[j];
                    arr[j] = arr[j+1];
                    arr[j+1] = temp;
                }
            }
        }
    }
    public static void main(String[] args){
        int[] numbers={9,2,6,3,8};
        bubbleSort(numbers);
        for(int num: numbers){
            System.out.print(num+" ");
        }
    }
    
}
//Time Complexity: O(n^2)
//Preferred when dataset is small / Nearly sorted

File: python/bubblesort.py
def bubble_sort(arr):
    n = len(arr)
    for i in range(n):
        # After each pass, the largest element moves to the end
        for j in range(0, n - i - 1):
            if arr[j] > arr[j + 1]:
                # Swap if elements are in wrong order
                arr[j], arr[j + 1] = arr[j + 1], arr[j]

# Example usage
numbers = [64, 34, 25, 12, 22, 11, 90]
print("Before sorting:", numbers)

bubble_sort(numbers)

print("After sorting:", numbers)
